Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -673,55 +725,112 @@ extension ChannelPipeline.SynchronousOperations {
     ///     and iterate over inbound HTTP/2 stream channels.
     @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-    @_spi(AsyncChannel)
-    public func configureAsyncHTTP2Pipeline<StreamInbound, StreamOutbound>(
+    internal func configureAsyncHTTP2Pipeline<StreamInbound, StreamOutbound>(
         mode: NIOHTTP2Handler.ParserMode,
-        connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration,
-        streamConfiguration: NIOHTTP2Handler.StreamConfiguration,
-        streamDelegate: NIOHTTP2StreamDelegate? = nil,
+        inboundStreamAsyncChannelConfiguration: NIOAsyncChannel<StreamInbound, StreamOutbound>.Configuration,
+        connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration = .init(),
+        streamConfiguration: NIOHTTP2Handler.StreamConfiguration = .init(),
         position: ChannelPipeline.Position = .last,
-        inboundStreamBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
-        isInboundStreamOutboundHalfClosureEnabled: Bool = false,
-        streamInboundType: StreamInbound.Type = StreamInbound.self,
-        streamOutboundType: StreamOutbound.Type = StreamOutbound.self,
-        inboundStreamInitializer: @escaping NIOHTTP2Handler.StreamInitializer
+        inboundStreamInitializer: @escaping NIOChannelInitializer
     ) throws -> NIOHTTP2Handler.AsyncStreamMultiplexer<NIOAsyncChannel<StreamInbound, StreamOutbound>> {
+        var configuration = NIOHTTP2Handler.Configuration()
+        configuration.connection = connectionConfiguration
+        configuration.stream = streamConfiguration
         return try self.configureAsyncHTTP2Pipeline(
             mode: mode,
-            connectionConfiguration: connectionConfiguration,
-            streamConfiguration: streamConfiguration,
-            streamDelegate: streamDelegate,
+            configuration: configuration,
             position: position
         ) { channel in
             inboundStreamInitializer(channel).flatMapThrowing { _ in
-                return try NIOAsyncChannel(
+                return try NIOAsyncChannel<StreamInbound, StreamOutbound>(
                     synchronouslyWrapping: channel,
-                    backpressureStrategy: inboundStreamBackpressureStrategy,
-                    isOutboundHalfClosureEnabled: isInboundStreamOutboundHalfClosureEnabled,
-                    inboundType: StreamInbound.self,
-                    outboundType: StreamOutbound.self
+                    backpressureStrategy: inboundStreamAsyncChannelConfiguration.backpressureStrategy,
+                    isOutboundHalfClosureEnabled: inboundStreamAsyncChannelConfiguration.isOutboundHalfClosureEnabled
                 )
             }
         }
     }
 }
+
+/// `NIONegotiatedHTTPVersion` is a generic negotiation result holder for HTTP/1.1 and HTTP/2
+@_spi(AsyncChannel)
+public enum NIONegotiatedHTTPVersion<HTTP1Output, HTTP2Output> {
```

Comment:
I think we require this:

```suggestion
public enum NIONegotiatedHTTPVersion<HTTP1Output: Sendable, HTTP2Output: Sendable>: Sendable {
```